### PR TITLE
chore: add more cdk errors in the error mapper

### PR DESCRIPTION
**Pull request by Amplifiyer**
## Changes

Extract more types of user errors from CDK log and map/classify them as `AmplifyErrors` to give less clutter in the sandbox log.

## Validation

unit tests

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/sandbox/src/sandbox_executor.ts [hunk 1]
```diff
@@ -64,7 +65,19 @@ export class AmplifySandboxExecutor {
   private getSecretLastUpdated = async (
     backendId: BackendIdentifier
   ): Promise<Date | undefined> => {
-    const secrets = await this.secretClient.listSecrets(backendId);
+    let secrets = undefined;
+    try {
+      secrets = await this.secretClient.listSecrets(backendId);
+    } catch (err) {
+      throw new AmplifyUserError(
```
Comment (edwardfoyle) [inline packages/sandbox/src/sandbox_executor.ts:72]: This should most likely be a fault

File: packages/backend-deployer/src/cdk_error_mapper.ts [hunk 2]
```diff
@@ -51,7 +51,25 @@ export class CdkErrorMapper {
       classification: 'ERROR',
     },
     {
-      errorRegex: /\[ERR_MODULE_NOT_FOUND\]:(.*)\n/,
+      errorRegex: /Unable to resolve AWS account to use/,
+      humanReadableErrorMessage:
+        'Unable to resolve AWS account to use. It must be either configured when you define your CDK Stack, or through the environment',
+      resolutionMessage:
+        'You can retry your last request as this is most likely a transient issue: https://github.com/aws/aws-cdk/issues/24744',
```
Comment (edwardfoyle) [inline packages/backend-deployer/src/cdk_error_mapper.ts:58]: I've definitely gotten this error when it was not transient (eg if I have stale local creds). I'd include something here about checking that local credentials are valid / refreshed.

File: packages/sandbox/src/sandbox_executor.ts (hunk 1 above)
Comment (Amplifiyer) [reply packages/sandbox/src/sandbox_executor.ts:72]: The only fault I saw here was expired credentials. Let me change this to look inside the `SecretError`.

Comment (edwardfoyle) [reply packages/sandbox/src/sandbox_executor.ts:72]: At the very least, a 5xx from the service is always possible. Things like throttling should also be faults, not errors

File: packages/backend-deployer/src/cdk_error_mapper.ts (hunk 2 above)
Comment (Amplifiyer) [reply packages/backend-deployer/src/cdk_error_mapper.ts:58]: I'll add it, but were you getting this exact `Unable to resolve AWS account to use` or `Unable to resolve AWS credentials`? From the mentioned github issue, it seems a widespread issue.

File: packages/sandbox/src/sandbox_executor.ts (hunk 1 above)
Comment (Amplifiyer) [reply packages/sandbox/src/sandbox_executor.ts:72]: yup, I changed it to fault except for one case where the underlying exception is `ExpiredTokenException`

File: packages/backend-deployer/src/cdk_error_mapper.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/backend-deployer/src/cdk_error_mapper.ts]: Ah maybe I was getting "Unable to resolve AWS credentials". Regardless, the new message looks good.

**Review by edwardfoyle — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**